tests/test_default_router.py

```diff
@@ -159,6 +159,10 @@ def test_invalid_field_name(self, template):
             ValueError,
             self.router.add_route, template, {}, ResourceWithId(-1))
 
+    def test_single_character_field_name(self):
+        self.router.add_route(
+            '/search?q={q}', {}, ResourceWithId(22))
```

Comment:
A quick question that just popped in my head. Was it intentional that you're using the field on a query param? Shouldn't the route just be `/search` and then access the query params via `req.params.get('q')`? The change is still relevant, but I just thought I would mention it in case you were looking for a way to get query parameters instead of route parameters.